pkg/cmd/server/origin/controller/config.go

```diff
@@ -244,9 +244,13 @@ func BuildOpenshiftControllerConfig(options configapi.MasterConfig) (*OpenshiftC
 		DefaultReplenishmentSyncPeriod: 12 * time.Hour,
 	}
 
-	// TODO this goes away with a truly generic autoscaler
+	// TODO: this goes away when we can get rid of the legacy Heapster metrics client
+	useRESTClientsOption := options.KubernetesMasterConfig.ControllerArguments["horizontal-pod-autoscaler-use-rest-clients"]
```

Comment:
You can't do this.  We don't want any dependencies from origin-controllers to kube-controllers.

Why would we allow this to be optional?  We can't just switch over unconditionally?

Rather than plumb our own set of flags through, can you undertake the work to make the upstream kube-controller-manager work for openshift?